server/src/main/java/com/cloud/network/router/VirtualNetworkApplianceManagerImpl.java

```diff
@@ -1290,15 +1285,15 @@ private void handleFailingChecks(DomainRouterVO router, List<String> failingChec
             s_logger.warn(alertMessage + ". Checking failed health checks to see if router needs reboot");
 
             String checkFailsToRecreateVr = RouterHealthChecksFailuresToRecreateVr.valueIn(router.getDataCenterId());
-            StringBuilder failingChecksEvent = new StringBuilder("Router " + router.getUuid() + " has failing checks:");
+            StringBuilder failingChecksEvent = new StringBuilder("Router " + router.getUuid() + " has failing checks: ");
             boolean recreateRouter = false;
             for (String failedCheck : failingChecks) {
-                failingChecksEvent.append(" ").append(failedCheck);
+                failingChecksEvent.append(failedCheck).append(", ");
                 if (StringUtils.isNotBlank(checkFailsToRecreateVr) && checkFailsToRecreateVr.contains(failedCheck)) {
                     recreateRouter = true;
                 }
             }
-
+            failingChecksEvent.setLength(failingChecksEvent.length() - 2);
```

Comment:
this is awkward. Can you append the ", " within the confines of the if?

and maybe make the "Router " + router.getUuid() + " has failing checks: " conditional on teh first check?

Reply:
Done.